### PR TITLE
feat: trending mints ep

### PR DESCRIPTION
**Pull request by r3lays**

### CODE REVIEW

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+  const conditions: string[] = [];
+  conditions.push(`kind = 'public'`, `status = 'open'`);
+  type && type !== "any" && conditions.push(`price ${type === "free" ? "= 0" : "> 0"}`);
+
+  const whereClause = conditions.length ? `WHERE ${conditions.join(" AND ")}` : "";
+
+  const baseQuery = `
+SELECT 
+    collection_id,
+    array_agg(
```
Comment (lunarmayor) [inline packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:268]: still feel like this is the wrong place to fetch this data. won't it mostly be thrown away?  you only need the mint stages for the final 50. maybe this is still more efficient even with that said. wdyt?

Comment (lunarmayor) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:268]: guess this does feel simpler for right now. alternative is the last collection fetch but that's getting cached

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+        ownerCount: Number(metadata.owner_count || 0),
+        floorAsk,
+      };
+    })
+  );
+
+  return collections;
+}
+
+async function getCollectionsMetadata(collectionIds: string[]): Promise<Record<string, Metadata>> {
```
Comment (lunarmayor) [inline packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:388]: wonder if should try to combine the trending collection function for this and this function. wouldn't it just be the same?

File: packages/indexer/src/elasticsearch/indexes/activities/index.ts [hunk 3]
```diff
@@ -343,6 +344,80 @@ export const getTopSellingCollections = async (params: {
...
+  contracts: string[];
+  startTime: number;
+  limit: number;
+}): Promise<ElasticMintResult[]> => {
+  const { contracts, startTime, limit } = params;
+
+  const salesQuery = {
+    bool: {
+      filter: [
+        {
```
Comment (lunarmayor) [inline packages/indexer/src/elasticsearch/indexes/activities/index.ts:357]: think we can use term here if it's just the one

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+            })
+              .allow(null)
+              .description("Lowest Ask Price."),
+          },
+          mintPrice: Joi.number().allow(null),
+          mintVolume: Joi.any(),
+          mintCount: Joi.number().allow(null),
+          mintType: Joi.string().allow("free", "paid", "", null),
+          mintStatus: Joi.string().allow("", null),
+          mintStages: Joi.array().items(
```
Comment (lunarmayor) [inline packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:189]: wonder if we should pull out more of the public mint stage like end and start, end, maxMints.

Comment (r3lays) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:189]: Yeah we can do that.

File: packages/indexer/src/elasticsearch/indexes/activities/index.ts (hunk 3 above)
Comment (r3lays) [reply packages/indexer/src/elasticsearch/indexes/activities/index.ts:357]: sg will fix.

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts (hunk 1 above)
Comment (r3lays) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:268]: Ill test the performance out to determine if we need the changes.

File: packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts (hunk 2 above)
Comment (r3lays) [reply packages/indexer/src/api/endpoints/collections/get-trending-mints/v1.ts:392]: In the trending collections we dont return the mint_stages.

**Pull request closed without merging**